### PR TITLE
support to get approximate keys and size for a given range from UserCollectedProperties

### LINKED ISSUE #11464: Support to get approximate keys and size from user collected properties
## Development Task

Currently approximate keys and size are only available for `KvEngine`, not for every tables. It's necessary to add interfaces to access them with `UserCollectedProperties`.

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: close #11464.
Extracted from #11385 .

### Check List 

Tests 

- No code

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by Connor1996 — APPROVED** [review]
LGTM

File: components/engine_rocks/src/table_properties.rs [hunk 1]
```diff
@@ -9,6 +9,12 @@ impl engine_traits::UserCollectedProperties for UserCollectedProperties {
     fn get(&self, index: &[u8]) -> Option<&[u8]> {
         self.0.get(index)
     }
+
+    fn approximate_size_and_keys(&self, start: &[u8], end: &[u8]) -> Option<(usize, usize)> {
+        let rp = RangeProperties::decode(&self.0).ok()?;
```
Comment (tabokie) [inline components/engine_rocks/src/table_properties.rs:14]: Where will this new method be called? This seems expensive.

**Review by tabokie — APPROVED** [review]

File: components/engine_rocks/src/table_properties.rs (hunk 1 above)
Comment (hicqu) [reply components/engine_rocks/src/table_properties.rs:14]: Here: https://github.com/tikv/tikv/pull/11385/files#r755931248.

**Pull request merged**